src/pages/[domain]/admin/configure-place.tsx

```diff
@@ -1,15 +1,246 @@
+  const { account, user, loading } = useContext(TalentLayerContext);
+  const chainId = useChainId();
+  const { data: walletClient } = useWalletClient({ chainId });
+  const { data: updateBuilderPlace, mutateAsync: updateBuilderPlaceAsync } =
+    useUpdateBuilderPlace();
+  const builderPlaceData = useGetBuilderPlaceFromOwner(user?.id as string);
+  const router = useRouter();
+  const [logoLoader, setLogoLoader] = useState(false);
+  const [logoErrorMessage, setLogoErrorMessage] = useState('');
+  const [palette, setPalette] = useState<iBuilderPlacePalette>(props.builderPlace?.palette);
+
+  const [colorName, setColorName] = useState('primary');
+  const [color, setColor] = useColor(palette[colorName as keyof iBuilderPlacePalette]);
+
+  const initialValues: IFormValues = {
+    subdomain: builderPlaceData?.subdomain || '',
+    logo: builderPlaceData?.logo || '',
+    palette,
+  };
+
+  useEffect(() => {
+    let timeoutId: string | number | NodeJS.Timeout | undefined;
+    // Prevents max stack depth being called
+    const delayedEffect = () => {
+      setPalette(prevPalette => {
+        return { ...prevPalette, [colorName]: color.hex };
+      });
+    };
+
+    timeoutId = setTimeout(delayedEffect, 10);
+    return () => clearTimeout(timeoutId);
+  }, [color]);
+
+  useEffect(() => {
+    document.documentElement.style.setProperty('--primary', palette.primary);
+    document.documentElement.style.setProperty('--primary-50', palette.primary + '60');
+    document.documentElement.style.setProperty('--primary-focus', palette.primaryFocus);
+    document.documentElement.style.setProperty('--primary-content', palette.primaryContent);
+    document.documentElement.style.setProperty('--base-100', palette.base100);
+    document.documentElement.style.setProperty('--base-200', palette.base200);
+    document.documentElement.style.setProperty('--base-300', palette.base300);
+    document.documentElement.style.setProperty('--base-content', palette.baseContent);
+    document.documentElement.style.setProperty('--info', palette.info);
+    document.documentElement.style.setProperty('--info-content', palette.infoContent);
+    document.documentElement.style.setProperty('--success', palette.success);
+    document.documentElement.style.setProperty('--success-50', palette.success + '60');
+    document.documentElement.style.setProperty('--success-content', palette.successContent);
+    document.documentElement.style.setProperty('--warning', palette.warning);
+    document.documentElement.style.setProperty('--warning-content', palette.warningContent);
+    document.documentElement.style.setProperty('--error', palette.error);
+    document.documentElement.style.setProperty('--error-content', palette.errorContent);
+  }, [palette]);
+
+  if (loading) {
+    console.log('no data');
+    return (
+      <div className='flex flex-col mt-5 pb-8'>
+        <Loading />
+      </div>
+    );
+  }
+
+  if (!builderPlaceData) {
```

Comment:
in this context no need

Reply:
Using serverside builderplace instead now